tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/IntArithPacker.scala

```diff
@@ -0,0 +1,56 @@
+package at.forsyte.apalache.tla.bmcmt.rules
+
+import at.forsyte.apalache.tla.bmcmt.{ArenaCell, SymbState, SymbStateRewriter}
+import at.forsyte.apalache.tla.lir.UntypedPredefs._
+import at.forsyte.apalache.tla.lir.oper.TlaArithOper
+import at.forsyte.apalache.tla.lir.values.TlaInt
+import at.forsyte.apalache.tla.lir.{NameEx, OperEx, ValEx}
+
+/**
+ * Rewrite integer arithmetic expressions, referring to arena cells for non-arithmetic subexpressions as necessary.
+ *
+ * @author
+ *   Thomas Pani
+ */
+trait IntArithPacker {
+  private lazy val substRule = new SubstRule
+
+  /**
+   * Rewrite [[state]]'s expression into an expression that is purely arithmetic, referring to arena cells for
+   * non-arithmetic subexpressions.
+   *
+   * @param state
+   *   current rewriter state, `state.ex` is the arithmetic expression to rewrite
+   * @param rewriter
+   *   rewriter to use for non-arithmetic subexpressions
+   * @return
+   *   the rewritten state
+   */
+  protected def packArithExpr(state: SymbState, rewriter: SymbStateRewriter): SymbState = state.ex match {
```

Comment:
```suggestion
  protected def packArithExpr(rewriter: SymbStateRewriter, state: SymbState): SymbState = state.ex match {
```

I usually collect the longer-living objects on the left. This is of course a matter of taste.